plugins/org.polymap.service.geoserver/src/org/polymap/service/geoserver/GeoServerServlet.java

```diff
@@ -177,9 +198,11 @@ public void init( ServletConfig config ) throws ServletException {
         context = new BundleServletContext( getServletContext() );
         log.info( "initGeoServer(): contextPath=" + context.getContextPath() );
         
-//        sessionKey = SessionContext.current().getSessionKey();
-//        assert sessionKey != null;
-
+        //String sessionKey = SessionContext.current().getSessionKey();
+        //assert sessionKey != null;
+        sessionKey = "ows_" + System.currentTimeMillis();
+        contextProvider.mapContext( sessionKey, true );
```

Comment:
Das verstehe ich nicht so richtig. Wo wird dieser SessionContext denn verwendet? Also, außer bei der Initialisierung? Er müsste in inContextClassLoader() gesetzt werden... aber wird nicht. Also läuft die Initialisierung in einem anderen Context als ein normaler Request?!

Wenn ein eigener SessionContext für GeoServer verwendet wird, dann muss auch beachtet werden, dass die einzelnen Entities auch wirklich aus diesem Context kommen. In diesem Fall mindestens #map.  

Außderdem bekommt man dann keine Änderungen der Entities in diesem separatem Context mit. Das wurde in SessionContext gemacht. Habe ich aber auch wegen der Komplexität auskommentiert. Also. es gibt ein paar Sachen zu beachten.

Ich war offenbar der Meinung ganz auf den SessionContext verzichten zu können, weil #map schon da ist und ansonsten nicht nochmal auf ein SessionSingleton zugegriffen wird....